### PR TITLE
Provide install targets

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Comment by TheOneRing** [comment]
@dschmidt @fmoc

**Comment by dschmidt** [comment]
Hm this probably also installs stuff when used as a submodule, right? 😏

**Comment by TheOneRing** [comment]
Hmm

> Hm this probably also installs stuff when used as a submodule, right? 😏

I think thats what exclude_from_all is for.
(Set in ownCloud)
https://cmake.org/cmake/help/latest/command/add_subdirectory.html

**Comment by fmoc** [comment]
`EXCLUDE_FROM_ALL` prevents targets from being pulled into the equivalent of `make all`. As long as targets are not referenced, they should not be installed. If you `target_link_libraries`, though, I think they will be `make install`ed.

**Comment by TheOneRing** [comment]
To be extra sure I tested it, and its indeed excluded.

File: src/libcrashreporter-gui/CMakeLists.txt [hunk 1]
```diff
@@ -38,5 +44,12 @@ target_link_libraries(crashreporter-gui
...
 )
 
+if(WIN32)
+    target_compile_definitions(crashreporter-gui PRIVATE UNICODE _UNICODE)
+endif()
+
 set_target_properties(crashreporter-gui PROPERTIES AUTOUIC ON)
 set_target_properties(crashreporter-gui PROPERTIES AUTOMOC ON)
+
+install(FILES CrashReporter.h DESTINATION ${INCLUDE_INSTALL_DIR}/CrashReporterQt/libcrashreporter-gui)
```
Comment (fmoc) [inline src/libcrashreporter-gui/CMakeLists.txt:54]: You may want to use `PUBLIC_HEADER` instead. That way you can add the header to the right target.

File: src/libcrashreporter-handler/CMakeLists.txt [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+if(WIN32)
+    target_compile_definitions(crashreporter-handler PRIVATE UNICODE _UNICODE)
+endif()
+
+target_include_directories(crashreporter-handler PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/..>
+        $<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}/CrashReporterQt>
+)
+
+install(FILES Handler.h DESTINATION ${INCLUDE_INSTALL_DIR}/CrashReporterQt/libcrashreporter-handler)
```
Comment (fmoc) [inline src/libcrashreporter-handler/CMakeLists.txt:21]: You may want to use `PUBLIC_HEADER` instead. That way you can add the header to the right target.

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/libcrashreporter-gui/CMakeLists.txt (hunk 1 above)
Comment (fmoc) [reply src/libcrashreporter-gui/CMakeLists.txt:54]: What's your concern? I've used `PUBLIC_HEADER` before to install public headers (on non-mac platforms, typically).

**Review by fmoc — APPROVED** [review]

**Pull request merged**